Explorer/Assets/DCL/PerformanceAndDiagnostics/Analytics/IAnalyticsController.cs

```diff
@@ -27,7 +29,16 @@ private sealed class NullAnalytics : IAnalyticsController
             public AnalyticsConfiguration Configuration => ScriptableObject.CreateInstance<AnalyticsConfiguration>();
 
             public void SetCommonParam(IRealmData _, IWeb3IdentityCache __, ExposedTransform ___) { }
+
             public void Track(string _, JsonObject __ = null) { }
         }
     }
+
+    public static class AnalyticsExtensions
+    {
+        public static IHealthCheck WithAnalytics(this IHealthCheck origin, IAnalyticsController analyticsController, string eventName, bool enableAnalytics) =>
+            enableAnalytics
+                ? new AnalyticsHealthCheckDecorator(origin, analyticsController, eventName)
+                : origin;
+    }
```

Comment:
As I understand, the main purpose of `AnalyticsHealthCheckDecorator` is to emit `AnalyticsEvent.LIVEKIT_HEALTH_CHECK_FAILED` event. Thoos, `eventName` parameter seems to be not needed (I can't imagine cases where mentioned class will use different event or will need to be constructed with different one).

Reply:
Secondly, `IHealthCheck` is very specific to Multiplayer. So I would suggest to move this nice builder inside  `AnalyticsHealthCheckDecorator` class and have it like that also for other ours Analytics decorator (with its own static builder method contained withing the class).  We can also think about some generic solution, but due to possibly much different set of parameters, generic solution could come quite overengineered...

Reply:
I'd like to move it there too, but it causes cycling asmdef exception

In case if hardcode event name inside to the class it should be renamed as FailAnalyticsHealthCheckDecorator

Reply:
I see. Maybe then we move it in a separate class and file with name `AnalyticsBuilder`?

Reply:
Good, renamed as FailAnalyticsHealthCheckDecorator